src/definitions.rs

```diff
@@ -4,62 +4,61 @@ use core::mem;
 #[repr(u8)]
 #[derive(Debug, Default, Copy, Clone)]
 pub enum State {
-    Anywhere = 0,
-    CsiEntry = 1,
-    CsiIgnore = 2,
-    CsiIntermediate = 3,
-    CsiParam = 4,
-    DcsEntry = 5,
-    DcsIgnore = 6,
-    DcsIntermediate = 7,
-    DcsParam = 8,
-    DcsPassthrough = 9,
-    Escape = 10,
-    EscapeIntermediate = 11,
+    CsiEntry,
+    CsiIgnore,
+    CsiIntermediate,
+    CsiParam,
+    DcsEntry,
+    DcsIgnore,
+    DcsIntermediate,
+    DcsParam,
+    DcsPassthrough,
+    Escape,
+    EscapeIntermediate,
+    OscString,
+    SosPmApcString,
+    Anywhere,
     #[default]
-    Ground = 12,
-    OscString = 13,
-    SosPmApcString = 14,
-    Utf8 = 15,
```

Comment:
There's only two places where UTF8 content can possibly appear, since escapes are generally ascii: In the ground state, so outside of the escape sequences and in OSCs.

Since the ground state is now handled by looking up the entire byte buffer and converting using `str::from_utf8`, we don't need a byte-by-byte parser for this anymore. And OSCs were never using the Utf8 state since we're just passing the OSC data as raw bytes.